TUM Campus App/TUMSexy/TUMSexyTableViewController.swift

```diff
@@ -72,26 +81,52 @@ final class TUMSexyTableViewController: UITableViewController {
     // MARK: UITableViewDataSource
     
     override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
-        return importer.fetchedResultsController.fetchedObjects?.count ?? 0
+        if searchController.isActive {
+            return filteredData.count
+        } else {
+            return importer.fetchedResultsController.fetchedObjects?.count ?? 0
+        }
     }
     
     override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
         let cell = tableView.dequeueReusableCell(withIdentifier: "TUMSexyLinkCell", for: indexPath)
-        let link = importer.fetchedResultsController.object(at: indexPath)
+    
+        if searchController.isActive {
+            cell.textLabel?.text = filteredData[indexPath.row].linkDescription
+        } else {
+            let link = importer.fetchedResultsController.object(at: indexPath)
+            cell.textLabel?.text = link.linkDescription
+        }
         
-        cell.textLabel?.text = link.linkDescription
         return cell
     }
 
     // MARK: UITableViewDelegate
 
     override func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
         tableView.deselectRow(at: indexPath, animated: true)
-        let link = importer.fetchedResultsController.object(at: indexPath)
+        let link: (TUMSexyLink)
+        
+        if searchController.isActive {
+            link = filteredData[indexPath.row]
+        } else {
+            link = importer.fetchedResultsController.object(at: indexPath)
+        }
+        
         guard let urlString = link.target, let url = URL(string: urlString) else { return }
         UIApplication.shared.open(url)
     }
     
-}
+    // MARK: UISearchResultsUpdating
+    func updateSearchResults(for searchController: UISearchController) {
+        filteredData.removeAll(keepingCapacity: false)
```

Comment:
Having two sources of truth is not that nice. I would suggest adding a `NSPredicate` to the `fetchedResultsController`. Something like this would be nice:

    // MARK: - UISearchResultsUpdating

    func updateSearchResults(for searchController: UISearchController) {
        guard let searchString = searchController.searchBar.text?.lowercased() else { return }
        let predicate = NSPredicate(format: "%K CONTAINS[cd] %@", #keyPath(TUMSexyLink.linkDescription), searchString)
        importer.fetchedResultsController.fetchRequest.predicate = predicate
        try? importer.fetchedResultsController.performFetch();
        self.tableView.reloadData()
    }